frontend/test/xray/xray.integ.spec.js

```diff
@@ -125,7 +140,132 @@ describe("xray integration tests", () => {
         })
     })
 
+    describe("admin management of xrays", async () => {
+        it("should allow an admin to manage xrays", async () => {
+            let app;
+
+            const store = await createTestStore()
+
+            store.pushPath('/admin/settings/x_rays')
+
+            app = mount(store.getAppContainer())
+
+            await store.waitForActions([LOAD_CURRENT_USER, INITIALIZE_SETTINGS])
+
+            const xraySettings = app.find(SettingsXrayForm)
+            const xrayToggle = xraySettings.find(Toggle)
+
+            // there should be a toggle
+            expect(xrayToggle.length).toEqual(1)
+
+            // things should be on
+            expect(getXrayEnabled(store.getState())).toEqual(true)
+            // the toggle should be on by default
+            expect(xrayToggle.props().value).toEqual(true)
+
+
+            // toggle the... toggle
+            click(xrayToggle)
+            await store.waitForActions([UPDATE_SETTING])
+
+            expect(getXrayEnabled(store.getState())).toEqual(false)
+
+
+            // navigate to a previosuly x-ray-able entity
+            store.pushPath(Urls.question(timeBreakoutQuestion.id()))
+            await store.waitForActions(INITIALIZE_QB, QUERY_COMPLETED)
+
+            // for some reason a delay is needed to get the full action suite
+            await delay(500);
+
+            const actionsWidget = app.find(ActionsWidget)
+            click(actionsWidget.childAt(0))
+
+            // there should not be an xray option
+            const xrayOptionIcon = actionsWidget.find('.Icon.Icon-beaker')
+            expect(xrayOptionIcon.length).toEqual(0)
+        })
+
+        it("should properly reflect the  an admin set the max cost of xrays", async () => {
+            await SettingsApi.put({ key: 'enable-xrays', value: true })
+            const store = await createTestStore()
+
+            store.pushPath('/admin/settings/x_rays')
+
+            const app = mount(store.getAppContainer())
+
+            await store.waitForActions([LOAD_CURRENT_USER, INITIALIZE_SETTINGS])
+
+            const xraySettings = app.find(SettingsXrayForm)
+
+            expect(xraySettings.find(Icon).length).toEqual(3)
+
+            const approximate = xraySettings.find('.text-measure li').first()
+
+            click(approximate)
+            await store.waitForActions([UPDATE_SETTING])
+
+            expect(approximate.hasClass('text-brand')).toEqual(true)
+            expect(getMaxCost(store.getState())).toEqual('approximate')
+
+            store.pushPath(`/xray/table/1/approximate`);
+
+            await store.waitForActions(FETCH_TABLE_XRAY, { timeout: 20000 })
+            await delay(200)
```

Comment:
you could consider increasing the delay to `500` or something (I did that in https://github.com/metabase/metabase/pull/5852 for many `delay`s and that seems to reduce the frequency of CI failures)